File: solutions/swift/rna-transcription/1/Sources/RnaTranscription/RnaTranscription.swift
//Solution goes in Sources

class RnaTranscription {
    enum TranscriptionError: Error {
        case invalidNucleotide(message: String)
    }
}

class Nucleotide {
    let rna: String
    let rnaToDna = ["G":"C", "C":"G", "T":"A", "A":"U"]
    
    init(_ rna: String) {
        self.rna = rna
    }
    
    func complementOfDNA() throws -> String {
        var ret = ""
        for char in rna {
            if let val = rnaToDna[String(char)] {
                ret += val
            } else {
                throw RnaTranscription.TranscriptionError.invalidNucleotide(message: "\(char) is not a valid Nucleotide")
            }
        }
        return ret
    }
}

File: solutions/csharp/rna-transcription/1/RnaTranscription.cs
public static class RnaTranscription
{
    public static string ToRna(string nucleotide)
    {
        var ret = "";
        foreach(var c in nucleotide)
        {
            switch (c) {
                case 'G': ret += "C"; break;
                case 'C': ret += "G"; break;
                case 'T': ret += "A"; break;
                case 'A': ret += "U"; break;
            }
        }
        return ret;
    }
}